Microsoft.Toolkit.Uwp.UI.Controls/AdaptiveGridView/AdaptiveGridView.cs

```diff
@@ -58,6 +65,13 @@ private void RecalculateLayout(double containerWidth)
             }
 
             ItemWidth = (containerWidth / _columns) - 5;
+
+            if (ItemAspectRatio > 0)
+            {
+                var safeRatio = Math.Min(100.0, Math.Max(0.01, ItemAspectRatio));
+                _templateProxy.Height = ItemWidth / safeRatio;
+                ItemHeight = _templateProxy.Height;
```

Comment:
A layout cycle shouldn't be updating the dependency properties. It's unexpected, could cause infinite layout cycles, and is probably why there's the code smell bit in the page sample.